src/python/pants/engine/isolated_process.py

```diff
@@ -133,6 +133,20 @@ class FallibleExecuteProcessResult:
     output_directory_digest: Digest
 
 
+@dataclass(frozen=True)
+class FallibleExecuteProcessResultWithPlatform:
+    """Result of executing a process.
+
+    Contains information about what platform a request ran on.
+    """
+
+    stdout: bytes
+    stderr: bytes
+    exit_code: int
+    output_directory_digest: Digest
+    platform: Platform
```

Comment:
It would work with FFI, it would just require more FFI boilerplate to set up, which is why I was disincentivized to do it that way. This is perhaps something worth addressing in our FFI code since I agree that your alternative is cleaner.